src/http_types.rs

```diff
@@ -129,6 +147,25 @@ impl<'a> From<&'a str> for Scheme {
     }
 }
 
+impl TryFrom<Scheme> for http::uri::Scheme {
+    type Error = InvalidScheme;
+
+    fn try_from(s: Scheme) -> Result<Self, Self::Error> {
+        match s.r#type.ok_or(InvalidScheme::MissingType)? {
+            scheme::Type::Registered(typ) => {
+                if typ == scheme::Registered::Http as i32 {
+                    Ok(http::uri::Scheme::HTTP)
+                } else if typ == scheme::Registered::Https as i32 {
+                    Ok(http::uri::Scheme::HTTPS)
+                } else {
+                    Err(InvalidScheme::Registered(typ))
+                }
+            }
```

Comment:
style nit, take it or leave it: we could collapse this into the top-level `match`?
```suggestion
            scheme::Type::Registered(typ) if typ == scheme::Registered::Http as i32 => Ok(http::uri::Scheme::HTTP),
            scheme::Type::Registered(typ) if typ == scheme::Registered::Https as i32 => Ok(http::uri::Scheme::HTTPS),
            scheme::Type::Registered(typ) => Err(InvalidScheme::Registered(typ)),
```